packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -621,7 +620,7 @@ class RenderSliverGrid extends RenderSliverMultiBoxAdaptor {
 
     final double paintExtent = calculatePaintOffset(
       constraints,
-      from: leadingScrollOffset,
+      from: math.min(constraints.scrollOffset.abs(), leadingScrollOffset),
```

Comment:
@chunhtai I think you are right, this should be the better way to fix the hitTest result. 🙂

Reply:
is the abs necessary? is it possible for the scroll offset to be come negative? even if it does we should still use the negative value to calculate the paint extent right?

Reply:
A negative scrollOffset happens when you over scroll at the top. When I was checking this yesterday there was a case where a negative scroll offset resulted in an incorrect paintExtent. Like in our example, instead of 500, when the scrollOffset was -10, it incorrectly calculated a paintExtent of 490. Checking now, it isn't reproducing, so maybe we do not need it.

Reply:
I took at calculatePaintOffset, if the scrollOffset is a negative number it should still correctly calculate the delta between from and to. I think we should remove the the abs if we cannot be reproduce it.

Reply:
sgtm!